### ISSUE TITLE
Issue with getting started

### ISSUE THREAD

**Issue by chand04** [gvegayon/parallel#27]
Hi,

I'm just trying out -parallel- and am probably running into a basic issue. I'm trying "Example 1" in your minimal examples and am getting the following. Is there something I'm missing?

Thanks,
Dave

. sysuse bplong.dta
(fictional blood-pressure data)
. sort patient
. parallel setclusters 2
N Clusters: 2
Stata dir:  /Applications/Stata/StataMP.app/Contents/MacOS/StataMP

.  parallel, by(patient): by patient: egen max_bp = max(bp)
--------------------------------------------------------------------------------
Parallel Computing with Stata (by GVY)
Clusters   : 2
pll_id     : 14un3xm8o1
Running at : /Users/davidchan/Dropbox/Research/ED peer effects/Data
Randtype   : datetime
Waiting for the clusters to finish...
cluster 0001 has finished with a connection error -601- (timeout) (see more)...
cluster 0002 has finished with a connection error -601- (timeout) (see more)...

Here are the results of creturn list:

. creturn list

System values

    --------------------------------------------------------------------------------------------------------------------
        c(current_date) = "22 Jan 2016"
        c(current_time) = "11:51:39"
           c(rmsg_time) = 0                          (seconds, from set rmsg)
    --------------------------------------------------------------------------------------------------------------------
       c(stata_version) = 14.1
             c(version) = 14.1                       (version)
         c(userversion) = 14.1                       (version)
    --------------------------------------------------------------------------------------------------------------------
           c(born_date) = "01 Dec 2015"
              c(flavor) = "IC"
                 c(bit) = 64
                  c(SE) = 1
                  c(MP) = 1
          c(processors) = 4                          (Stata/MP, set processors)
      c(processors_lic) = 8
     c(processors_mach) = 4
      c(processors_max) = 4
                c(mode) = ""
             c(console) = ""
    --------------------------------------------------------------------------------------------------------------------
                  c(os) = "MacOSX"
               c(osdtl) = "10.11.3"
            c(hostname) = "DN0a22e56a.SUNet"
        c(machine_type) = "Macintosh (Intel .."
           c(byteorder) = "lohi"
            c(username) = "davidchan"
    --------------------------------------------------------------------------------------------------------------------

Directories and paths

    --------------------------------------------------------------------------------------------------------------------
        c(sysdir_stata) = "/Applications/Sta.."      (sysdir)
         c(sysdir_base) = "/Applications/Sta.."      (sysdir)
         c(sysdir_site) = "/Applications/Sta.."      (sysdir)
         c(sysdir_plus) = "~/Library/Applica.."      (sysdir)
     c(sysdir_personal) = "~/Library/Applica.."      (sysdir)
     c(sysdir_oldplace) = "~/ado/"                   (sysdir)
              c(tmpdir) = "/var/folders/zt/2.."
    --------------------------------------------------------------------------------------------------------------------
             c(adopath) = "BASE;SITE;.;PERSO.."      (adopath)
                 c(pwd) = "/Users/davidchan/.."      (cd)
              c(dirsep) = "/"
    --------------------------------------------------------------------------------------------------------------------

System limits

    --------------------------------------------------------------------------------------------------------------------
        c(max_N_theory) = 281474976710655
        c(max_k_theory) = 5000                       (set maxvar)
    c(max_width_theory) = 1048576                    (set maxvar)
    --------------------------------------------------------------------------------------------------------------------
         c(max_matsize) = 11000
         c(min_matsize) = 10
    --------------------------------------------------------------------------------------------------------------------
        c(max_macrolen) = 4227143
            c(macrolen) = 645200                     (set maxvar)
             c(charlen) = 67783
          c(max_cmdlen) = 4227159
              c(cmdlen) = 645216                     (set maxvar)
         c(namelenbyte) = 128
         c(namelenchar) = 32
               c(eqlen) = 1337
    --------------------------------------------------------------------------------------------------------------------

Numerical and string limits

    --------------------------------------------------------------------------------------------------------------------
           c(mindouble) = -8.9884656743e+307
           c(maxdouble) = 8.9884656743e+307
           c(epsdouble) = 2.22044604925e-16
      c(smallestdouble) = 2.2250738585e-308
    --------------------------------------------------------------------------------------------------------------------
            c(minfloat) = -1.70141173319e+38
            c(maxfloat) = 1.70141173319e+38
            c(epsfloat) = 1.19209289551e-07
    --------------------------------------------------------------------------------------------------------------------
             c(minlong) = -2147483647
             c(maxlong) = 2147483620
    --------------------------------------------------------------------------------------------------------------------
              c(minint) = -32767
              c(maxint) = 32740
    --------------------------------------------------------------------------------------------------------------------
             c(minbyte) = -127
             c(maxbyte) = 100
    --------------------------------------------------------------------------------------------------------------------
        c(maxstrvarlen) = 2045
       c(maxstrlvarlen) = 2000000000
        c(maxvlabellen) = 32000
    --------------------------------------------------------------------------------------------------------------------

Current dataset

    --------------------------------------------------------------------------------------------------------------------
                   c(N) = 240
                   c(k) = 5
               c(width) = 7
             c(changed) = 0
            c(filename) = "/Applications/Sta.."
            c(filedate) = "22 Jan 2016 11:47"
    --------------------------------------------------------------------------------------------------------------------

Memory settings

    --------------------------------------------------------------------------------------------------------------------
              c(memory) = 33554432
              c(maxvar) = 5000                       (set maxvar)
             c(matsize) = 400                        (set matsize)
            c(niceness) = 5                          (set min_memory)
          c(min_memory) = 0                          (set min_memory)
          c(max_memory) = .                          (set max_memory)
         c(segmentsize) = 33554432                   (set segmentsize)
    --------------------------------------------------------------------------------------------------------------------

Output settings

    --------------------------------------------------------------------------------------------------------------------
                c(more) = "on"                       (set more)
                c(rmsg) = "off"                      (set rmsg)
                  c(dp) = "period"                   (set dp)
            c(linesize) = 120                        (set linesize)
            c(pagesize) = 28                         (set pagesize)
             c(logtype) = "smcl"                     (set logtype)
             c(noisily) = 1
             c(eolchar) = "unix"                     (set eolchar)
          c(notifyuser) = "on"                       (set notifyuser)
             c(playsnd) = "off"                      (set playsnd)
      c(include_bitmap) = "on"                       (set include_bitmap)
    --------------------------------------------------------------------------------------------------------------------
               c(level) = 95                         (set level)
              c(clevel) = 95                         (set clevel)
    --------------------------------------------------------------------------------------------------------------------
      c(showbaselevels) = ""                         (set showbaselevels)
      c(showemptycells) = ""                         (set showemptycells)
         c(showomitted) = ""                         (set showomitted)
             c(fvlabel) = "on"                       (set fvlabel)
              c(fvwrap) = 1                          (set fvwrap)
            c(fvwrapon) = "word"                     (set fvwrapon)
            c(lstretch) = ""                         (set lstretch)
    --------------------------------------------------------------------------------------------------------------------
             c(cformat) = ""                         (set cformat)
             c(sformat) = ""                         (set sformat)
             c(pformat) = ""                         (set pformat)
    --------------------------------------------------------------------------------------------------------------------
      c(coeftabresults) = "on"                       (set coeftabresults)

Interface settings

    --------------------------------------------------------------------------------------------------------------------
          c(reventries) = 5000                       (set reventries)
         c(revkeyboard) = "on"                       (set revkeyboard)
         c(varkeyboard) = "on"                       (set varkeyboard)
         c(smoothfonts) = "on"                       (set smoothfonts)
             c(linegap) = 1                          (set linegap)
       c(scrollbufsize) = 204800                     (set scrollbufsize)
               c(maxdb) = 50                         (set maxdb)
    --------------------------------------------------------------------------------------------------------------------

Graphics settings

    --------------------------------------------------------------------------------------------------------------------
            c(graphics) = "on"                       (set graphics)
              c(scheme) = "s2color"                  (set scheme)
          c(printcolor) = "automatic"                (set printcolor)
           c(copycolor) = "automatic"                (set copycolor)
    --------------------------------------------------------------------------------------------------------------------

Efficiency settings

    --------------------------------------------------------------------------------------------------------------------
             c(adosize) = 1000                       (set adosize)
    --------------------------------------------------------------------------------------------------------------------

Network settings

    --------------------------------------------------------------------------------------------------------------------
            c(checksum) = "off"                      (set checksum)
            c(timeout1) = 30                         (set timeout1)
            c(timeout2) = 180                        (set timeout2)
    --------------------------------------------------------------------------------------------------------------------
           c(httpproxy) = "off"                      (set httpproxy)
       c(httpproxyhost) = ""                         (set httpproxyhost)
       c(httpproxyport) = 80                         (set httpproxyport)
    --------------------------------------------------------------------------------------------------------------------
       c(httpproxyauth) = "off"                      (set httpproxyauth)
       c(httpproxyuser) = ""                         (set httpproxyuser)
         c(httpproxypw) = ""                         (set httpproxypw)
    --------------------------------------------------------------------------------------------------------------------

Update settings

    --------------------------------------------------------------------------------------------------------------------
        c(update_query) = "on"                       (set update_query)
     c(update_interval) = 7                          (set update_interval)
       c(update_prompt) = "on"                       (set update_prompt)
    --------------------------------------------------------------------------------------------------------------------

Trace (program debugging) settings

    --------------------------------------------------------------------------------------------------------------------
               c(trace) = "off"                      (set trace)
          c(tracedepth) = 32000                      (set tracedepth)
            c(tracesep) = "on"                       (set tracesep)
         c(traceindent) = "on"                       (set traceindent)
         c(traceexpand) = "on"                       (set traceexpand)
         c(tracenumber) = "off"                      (set tracenumber)
         c(tracehilite) = ""                         (set tracehilite)
    --------------------------------------------------------------------------------------------------------------------

Mata settings

    --------------------------------------------------------------------------------------------------------------------
          c(matastrict) = "off"                      (set matastrict)
            c(matalnum) = "off"                      (set matalnum)
        c(mataoptimize) = "on"                       (set mataoptimize)
           c(matafavor) = "space"                    (set matafavor)
           c(matacache) = 2000                       (set matacache)
            c(matalibs) = "lmatabase;lmataad.."      (set matalibs)
         c(matamofirst) = "off"                      (set matamofirst)
    --------------------------------------------------------------------------------------------------------------------

Unicode settings

    --------------------------------------------------------------------------------------------------------------------
           c(locale_ui) = "en_US"                    (set locale_ui)
    c(locale_functions) = "en_US"                    (set locale_functions)
      c(locale_icudflt) = "en_US"                    (unicode locale)
    --------------------------------------------------------------------------------------------------------------------

Other settings

    --------------------------------------------------------------------------------------------------------------------
                c(type) = "float"                    (set type)
             c(maxiter) = 16000                      (set maxiter)
       c(searchdefault) = "all"                      (set searchdefault)
                 c(rng) = "default"                  (set rng)
         c(rng_current) = "mt64"
            c(rngstate) = "XAAe85dd500d27408.."      (set rngstate)
           c(varabbrev) = "on"                       (set varabbrev)
          c(emptycells) = "keep"                     (set emptycells)
             c(odbcmgr) = "iodbc"                    (set odbcmgr)
          c(odbcdriver) = "unicode"                  (set odbcdriver)
    --------------------------------------------------------------------------------------------------------------------

Other

    --------------------------------------------------------------------------------------------------------------------
                  c(pi) = 3.141592653589793
               c(alpha) = "a b c d e f g h i.."
               c(ALPHA) = "A B C D E F G H I.."
                c(Mons) = "Jan Feb Mar Apr M.."
              c(Months) = "January February .."
               c(Wdays) = "Sun Mon Tue Wed T.."
            c(Weekdays) = "Sunday Monday Tue.."
                  c(rc) = 0                          (capture)
    --------------------------------------------------------------------------------------------------------------------

**Comment by bquistorff** [comment]
Works for me on Windows 10 Stata 11.

**Comment by gvegayon** [comment]
Seems to me that there's an issue finding stata bin. Can you check if

```
Stata dir: /Applications/Stata/StataMP.app/Contents/MacOS/StataMP
```

exists?

Thanks

**Comment by chand04** [comment]
Hi all,

Thanks for your response. That directory does not exist.

Thanks,
Dave

**Comment by gvegayon** [comment]
That's the issue then. Parallel tries to find the executable file ("Stata.exe") on your computer. If it fails you can always point directly to it using the `statapath` option, for example

```
parallel setclusters 4, statapath(c:/stata/stata.exe)
```

You can do the same thing on Mac (obviously replacing the argument in `statapath`

Best,

**Comment by bquistorff** [comment]
Note that it can be hard to tell if a folder exists on MacOS since application extensions (`.app`) are usually hidden (so inside `/Applications/Stata/` there might just be `StataMP`) and to get inside the Stata package from the finder you need to `Ctrl` click the package and choose `Show package contents`.

**Comment by chand04** [comment]
That's really helpful Brian. I did that, and now I'm going to have to
revise my response. That directory does exist.

Thanks,
Dave

**Comment by bquistorff** [comment]
Has this been resolved? If not, what are the contents of the Stata directory mentioned? I tried this example with Stata 13.1 on MacOS X and it worked fine. Maybe you have a bit of a weird setup.

**Comment by bquistorff** [comment]
Closing for now. Reopen if needbe.

**Comment by chand04** [comment]
Thanks Brian, and sorry for not responding sooner. I haven't been able to
follow up on this, but I'll let you know if I'm still having issues.

Thanks
Dave

**Comment by chand04** [comment]
Hi all,

I just tried this again, and I'm still getting the same error. I verified
that

Stata dir: /Applications/Stata/StataMP.app/Contents/MacOS/StataMP

exists.

Thanks,
Dave

**Comment by bquistorff** [comment]
Are you using the latest version?

**Comment by chand04** [comment]
Thanks for your reply Brian. How would I install the latest version? I just
tried typing "ssc install parallel" again, and got this message:

.  ssc install parallel
checking parallel consistency and verifying not already installed...

the following files already exist and are different:
    /Users/davidchan/Library/Application
Support/Stata/ado/plus/p/parallel_source.sthlp
    /Users/davidchan/Library/Application
Support/Stata/ado/plus/l/lparallel.mlib

no files installed or copied
(no action taken)


Thanks,
Dave

**Comment by gvegayon** [comment]
Hi, try following these instructions:

https://github.com/gvegayon/parallel#development-version-latestmaster

Best,

George G. Vega Yon
+1 (626) 381 8171
http://cana.usc.edu/vegayon

**Comment by gvegayon** [comment]
This was actually a bug part of Stata and it was fixed (on our side) on #47
